### PR TITLE
feat(docz-components): add fullscreen mode

### PR DESCRIPTION
**Pull request by jesperorb**
### Description

Add a fullscreen mode to `Playground`

### Features

- [X] Modal + button to open modal 
- [X] Close on escape
- [X] Close on click outside
- [X] Styling inside of modal (take up whole modal)
- [x] Use theme UI variables for styling (ability to mod the dialog from themeConfig)
- [x] Quick select viewport (mobile, tablet, desktop)

### Screenshots

Kapture 2019-10-06 at 0 40 37

### CODE REVIEW

**Comment by Dangoo** [comment]
@jesperorb shouldn‘t the full screen mode remove the whole chrome of the Playground for actual responsive testing and to avoid sideeffects?

**Comment by jesperorb** [comment]
> @jesperorb shouldn‘t the full screen mode remove the whole chrome of the Playground for actual responsive testing and to avoid sideeffects?

What do you mean by "chrome of the Playground"?

File: core/docz-components/src/components/Playground/index.tsx [hunk 1]
```diff
@@ -90,27 +139,21 @@ export const Playground = ({
...
+        {editor}
+        {showFullscreen && (
+          <Dialog
+            title="Live Preview"
+            onClose={toggleFullscreen}
+            data-testid="livepreview-fullscreen"
+            width={dialogWidth}
+          >
+            <DialogActions onChangeSize={onChangeSize} />
+            {preview}
```
Comment (rakannimer) [inline core/docz-components/src/components/Playground/index.tsx]: Shouldn't we also have `error` and `editor` here ?

Comment (jesperorb) [reply core/docz-components/src/components/Playground/index.tsx]: Sure, can add it

File: core/docz-components/src/utils/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      if(document !== undefined) {
+        document.body.style.overflow = 'unset'
+      }
+    }
+  }, []);
+}
+
+export const useEscape = (callback: () => void): void => {
+  const keydownCallback = (event: any): void => {
+    if(event.keyCode === 27){
```
Comment (mickaelzhang) [inline core/docz-components/src/utils/hooks.tsx]: `keyCode` has been deprecated, you can use `key` instead

File: core/docz-components/src/utils/hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      callback();
+    }
+    return
+  }
+  useEffect(() => {
+    if(document !== undefined){
```
Comment (mickaelzhang) [inline core/docz-components/src/utils/hooks.tsx]: We are specifying the condition 2 time here, maybe we could do something like:

```ts
useEffect(() => {
  if (document === undefined) {
    return () => {}
  }

  document.addEventListener('keydown', keydownCallback, true)
  return () => document.removeEventListener('keydown', keydownCallback, true)
})
```

I find it more clear but might be personal preference 👍

File: core/docz-components/src/utils/hooks.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { useEffect } from 'react'
+
+export const useScrollLock = (): void => {
+  useEffect(() => {
```
Comment (mickaelzhang) [inline core/docz-components/src/utils/hooks.tsx]: Same comment as `useEscape` concerning the condition

**Comment by chaosmirage** [comment]
Can this PR be merged?

**Comment by Bhavin789** [comment]
@jesperorb Are there any plans to reopen this PR? We have been seeing very long playgrounds lately and it becomes too difficult to scroll on the whole page. Even a handy show/collapse will help. Thanks!

**Pull request closed without merging**